### PR TITLE
Allow load_earth_relief() to load pixel or gridline registered data

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Somewhat backward compatible way of retrieving the `earth_relief` grids using GMT 6.0 or GMT 6.1. Tests on GMT 6.0 should still pass as usual in this PR, but there should be fewer failures on GMT 6.1 (except those due to the cpt change).

TODO:
- [x] Modify earth_relief.py to allow for getting pixel/gridline earth_relief grids (80170245ac2e781468dd52166a993ce9866faefe) **53** test failures on GMT 6.1
- [x] Change unit tests to use `load_earth_relief(pixel_reg=False)` (9a7491059906c7c37f4c352e7abfb16908119f1e) **39** test failures on GMT 6.1

Fixes (partially) #489

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: pygmt/datasets/earth_relief.py [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 import xarray as xr
+from packaging.version import Version
 
-from .. import which
+from .. import clib, which
 from ..exceptions import GMTInvalidInput
 
 
-def load_earth_relief(resolution="01d"):
+def load_earth_relief(resolution="01d", pixel_reg=None):
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:12]: I personally prefer an argument like `registration="pixel"` or `registration="gridline"`, just like **-r** option in most GMT modules.

Comment (EJFielding) [reply pygmt/datasets/earth_relief.py:12]: I agree that `registration="pixel"` is much better.

Comment (weiji14) [reply pygmt/datasets/earth_relief.py:12]: Yep, this can be arranged.

**Comment by seisman** [comment]
@weiji14 Do you want to implement the registration option only, or do you also want to support `region` (by using `grdcut` in this PR?

**Comment by weiji14** [comment]
Let's keep this PR small. I actually have no idea how the `grdcut` and `which` commands go together, might be better for you to handle it.

**Comment by seisman** [comment]
OK. Please merge the master branch in then I can review it.

File: pygmt/datasets/earth_relief.py [hunk 2]
```diff
@@ -28,6 +29,11 @@ def load_earth_relief(resolution="01d"):
         ``'20m'``, ``'15m'``, ``'10m'``, ``'06m'``, ``'05m'``, ``'04m'``,
         ``'03m'``, ``'02m'``, ``'01m'``, ``'30s'`` or ``'15s'``.
 
+    registration : str
+        Grid registration type. Either ``pixel`` for pixel registration or
+        ``gridline`` for gridline registration. Default is ``None``, which
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:34]: See https://docs.generic-mapping-tools.org/latest/datasets/earth-relief.html.

> If *reg* is not specified we will return the pixel-registered version unless only the gridline-registered file is available.

File: pygmt/datasets/earth_relief.py [hunk 3]
```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
...
+        _registration = ""  # If None, let GMT decide on Pixel/Gridline type
+        with clib.Session() as lib:
+            if registration is not None and Version(lib.info["version"]) >= Version(
+                "6.1.0"
+            ):
+                _registration = f"_{registration[0]}"
+    else:
+        raise GMTInvalidInput(
+            f"Invalid grid registration: {registration}, should be either "
+            "'pixel', 'gridline' or None. Default is None for pixel registration."
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:56]: Same here.

File: pygmt/datasets/earth_relief.py [hunk 4]
```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
 
     """
     _is_valid_resolution(resolution)
-    fname = which("@earth_relief_{}".format(resolution), download="u")
+
+    if registration in ("pixel", "gridline", None):
+        _registration = ""  # If None, let GMT decide on Pixel/Gridline type
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:47]: The variable name `_registration` is a little confusing here. What about *reg*?

File: pygmt/datasets/earth_relief.py [hunk 5]
```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
...
+        with clib.Session() as lib:
+            if registration is not None and Version(lib.info["version"]) >= Version(
+                "6.1.0"
+            ):
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:51]: I think this is also good:
```suggestion
            if registration and Version(lib.info["version"]) >= Version("6.1.0"):
```

**Comment by weiji14** [comment]
Oops, should have used `mkdir -p ~/.gmt` instead of `mkdir ~/.gmt` at https://github.com/GenericMappingTools/pygmt/blob/a5af2dc5ecca2a02dc8bacb24f8d55d56c330e4f/.github/workflows/ci_tests.yaml#L72

Should I just fix it here?

File: pygmt/datasets/earth_relief.py [hunk 6]
```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
...
+                _registration = f"_{registration[0]}"
+    else:
+        raise GMTInvalidInput(
+            f"Invalid grid registration: {registration}, should be either "
+            "'pixel', 'gridline' or None. Default is None for pixel registration."
+        )
+
+    fname = which(f"@earth_relief_{resolution}{_registration}", download="u")
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:59]: It's better to use `download="a"` in GMT 6.1, but it's OK now.

File: pygmt/tests/test_datasets.py [hunk 7]
```diff
@@ -80,9 +80,15 @@ def test_earth_relief_01d():
```
Comment (seisman) [inline pygmt/tests/test_datasets.py:80]: After bumping to GMT 6.1, we need to add a test with `registration="pixel"` or None.

**Comment by seisman** [comment]
> Oops, should have used `mkdir -p ~/.gmt` instead of `mkdir ~/.gmt` at
> 
> https://github.com/GenericMappingTools/pygmt/blob/a5af2dc5ecca2a02dc8bacb24f8d55d56c330e4f/.github/workflows/ci_tests.yaml#L72
> 
> Should I just fix it here?

It's a little weird. For the ubuntu job, I saw
```
Cache restored from key: cache-gmt-refs/heads/master-macOS-20200609
```
why macOS?

**Comment by weiji14** [comment]
> It's a little weird. For the ubuntu job, I saw
> 
> ```
> Cache restored from key: cache-gmt-refs/heads/master-macOS-20200609
> ```
> 
> why macOS?

Ah, we'll need to append the matrix-os to the restore-from key then. Maybe I should open a separate PR :facepalm:

**Comment by seisman** [comment]
> > It's a little weird. For the ubuntu job, I saw
> > ```
> > Cache restored from key: cache-gmt-refs/heads/master-macOS-20200609
> > ```
> > 
> > 
> > why macOS?
> 
> Ah, we'll need to append the matrix-os to the restore-from key then. Maybe I should open a separate PR 🤦

I see. Please open another PR instead.

**Comment by seisman** [comment]
Also, `mkdir -p` is better.

**Comment by seisman** [comment]
BTW, the PR title can be changed to "Allow load_earth_relief() to load pixel or gridline registered data". It's more readable to me.

File: pygmt/datasets/earth_relief.py [hunk 8]
```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
...
     _is_valid_resolution(resolution)
-    fname = which("@earth_relief_{}".format(resolution), download="u")
+
+    if registration in ("pixel", "gridline", None):
+        _registration = ""  # If None, let GMT decide on Pixel/Gridline type
+        with clib.Session() as lib:
+            if registration is not None and Version(lib.info["version"]) >= Version(
+                "6.1.0"
+            ):
+                _registration = f"_{registration[0]}"
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:52]: If GMT<6.1.0, we should raise an warning like "The 'registration' argument is only available for GMT>=6.1.0".

File: pygmt/datasets/earth_relief.py (hunk 2 above)
Comment (weiji14) [reply pygmt/datasets/earth_relief.py]: Done.

File: pygmt/datasets/earth_relief.py (hunk 8 above)
Comment (weiji14) [reply pygmt/datasets/earth_relief.py]: `registration="gridline"` will work on GMT 6.0, so I'll just raise `GMTInvalidInput` when GMT 6.0 users try to pass in `registration="pixel"`.
